wp-seo-main.php

```diff
@@ -27,6 +27,53 @@
 	define( 'WPSEO_CSSJS_SUFFIX', ( ( defined( 'SCRIPT_DEBUG' ) && true === SCRIPT_DEBUG ) ? '' : '.min' ) );
 }
 
+global $pagenow;
+if ( is_admin() && $pagenow === 'plugins.php' ) {
+	add_action( 'admin_init', 'yoast_seo_free_register_deactivate_listener' );
+	add_action( 'admin_init', 'yoast_seo_premium_register_deactivate_listener' );
+}
+
+/**
+ * Listener for Yoast SEO free. Disable Free when someone is trying to activate Premium.
+ */
+function yoast_seo_free_register_deactivate_listener() {
+	if ( ( $listener_plugin = filter_input( INPUT_GET, 'plugin' ) ) && $listener_plugin === 'wordpress-seo-premium/wp-seo-premium.php' ) {
+		wp_redirect( admin_url( 'plugins.php?yoast_seo_disable=free&_yoast_disable_nonce=' . wp_create_nonce( 'yoast-seo-disable-nonce' ) ), 301 );
+		exit;
+	}
+
+	if ( ( $listener_plugin = filter_input( INPUT_GET, 'yoast_seo_disable' ) ) && $listener_plugin === 'free' && ( $listener_nonce = filter_input( INPUT_GET, '_yoast_disable_nonce' ) ) && wp_verify_nonce( $listener_nonce, 'yoast-seo-disable-nonce' ) ) {
+		deactivate_plugins( 'wordpress-seo/wp-seo.php' );
+
+		Yoast_Notification_Center::get()->add_notification(
+			new Yoast_Notification( __( 'We\'ve disabled the free Yoast SEO plugin and activated the Yoast Premium SEO plugin.', 'wordpress-seo' ), array( 'type' => 'updated' ) )
+		);
+
+		wp_redirect( admin_url( 'plugins.php?action=activate&plugin=wordpress-seo-premium%2Fwp-seo-premium.php&plugin_status=all&paged=1&s&_wpnonce=' . wp_create_nonce( 'activate-plugin_wordpress-seo-premium/wp-seo-premium.php' ) ) );
+		exit;
+	}
+}
+
+/**
+ * Listener for Yoast SEO premium. Disable premium when someone is trying to activate free.
+ */
+function yoast_seo_premium_register_deactivate_listener() {
+	if ( ( $listener_plugin = filter_input( INPUT_GET, 'plugin' ) ) && $listener_plugin === 'wordpress-seo/wp-seo.php' ) {
+		wp_redirect( admin_url( 'plugins.php?yoast_seo_disable=premium&_yoast_disable_nonce=' . wp_create_nonce( 'yoast-seo-disable-nonce' ) ), 301 );
+		exit;
+	}
+
+	if ( ( $listener_plugin = filter_input( INPUT_GET, 'yoast_seo_disable' ) ) && $listener_plugin === 'premium' && ( $listener_nonce = filter_input( INPUT_GET, '_yoast_disable_nonce' ) ) && wp_verify_nonce( $listener_nonce, 'yoast-seo-disable-nonce' ) ) {
```

Comment:
Why not: `wp_verify_nonce( $filter_input( INPUT_GET, '_yoast_disable_nonce' ), 'yoast-seo-disable-nonce' )` There is no need to store the nonce in its own variable.